src/dialects/ibmi/query-generator-typescript.ts

```diff
@@ -0,0 +1,44 @@
+import { joinSQLFragments } from '../../utils/join-sql-fragments';
+import { AbstractQueryGenerator } from '../abstract/query-generator';
+import type { TableNameWithSchema } from '../abstract/query-interface';
+
+/**
+ * Temporary class to ease the TypeScript migration
+ */
+export class IBMiQueryGeneratorTypeScript extends AbstractQueryGenerator {
+  describeTableQuery(table: TableNameWithSchema) {
+    return joinSQLFragments([
+      'SELECT',
+      'QSYS2.SYSCOLUMNS.*,',
+      'QSYS2.SYSCST.CONSTRAINT_NAME,',
+      'QSYS2.SYSCST.CONSTRAINT_TYPE',
+      'FROM QSYS2.SYSCOLUMNS',
+      'LEFT OUTER JOIN QSYS2.SYSCSTCOL',
+      'ON QSYS2.SYSCOLUMNS.TABLE_SCHEMA = QSYS2.SYSCSTCOL.TABLE_SCHEMA',
+      'AND QSYS2.SYSCOLUMNS.TABLE_NAME = QSYS2.SYSCSTCOL.TABLE_NAME',
+      'AND QSYS2.SYSCOLUMNS.COLUMN_NAME = QSYS2.SYSCSTCOL.COLUMN_NAME',
+      'LEFT JOIN QSYS2.SYSCST',
+      'ON QSYS2.SYSCSTCOL.CONSTRAINT_NAME = QSYS2.SYSCST.CONSTRAINT_NAME',
+      'WHERE QSYS2.SYSCOLUMNS.TABLE_SCHEMA =',
+      table.schema !== this.dialect.getDefaultSchema() ? `${this.escape(table.schema)}` : 'CURRENT SCHEMA',
+      'AND QSYS2.SYSCOLUMNS.TABLE_NAME =',
+      `${this.escape(table.tableName)}`,
+    ]);
+  }
+
+  showIndexesQuery(table: TableNameWithSchema) {
+    // TODO [+odbc]: check if the query also works when capitalized (for consistency)
+    return joinSQLFragments([
+      'select QSYS2.SYSCSTCOL.CONSTRAINT_NAME as NAME, QSYS2.SYSCSTCOL.COLUMN_NAME, QSYS2.SYSCST.CONSTRAINT_TYPE, QSYS2.SYSCST.TABLE_SCHEMA,',
+      'QSYS2.SYSCST.TABLE_NAME from QSYS2.SYSCSTCOL left outer join QSYS2.SYSCST on QSYS2.SYSCSTCOL.TABLE_SCHEMA = QSYS2.SYSCST.TABLE_SCHEMA and',
+      'QSYS2.SYSCSTCOL.TABLE_NAME = QSYS2.SYSCST.TABLE_NAME and QSYS2.SYSCSTCOL.CONSTRAINT_NAME = QSYS2.SYSCST.CONSTRAINT_NAME where',
+      'QSYS2.SYSCSTCOL.TABLE_SCHEMA =',
+      table.schema !== this.dialect.getDefaultSchema() ? `${this.escape(table.schema)}` : 'CURRENT SCHEMA',
```

Comment:
same here. Default to CURRENT SCHEMA only if the schema is blank - as a temporary workaround until we fix IBMI's getDefaultSchema